src/workers/event.rs

```diff
@@ -741,17 +769,44 @@ where
     async fn run_all_scalers(&self, event: &Event) -> anyhow::Result<()> {
         let scalers = self.scalers.get_all_scalers().await;
 
-        let (commands, res) = get_commands_and_result(
-            scalers
-                .values()
-                .flatten()
-                .map(|scaler| scaler.handle_event(event)),
-            "Errors occurred while handling event with all scalers",
-        )
-        .await;
+        let futs = scalers.iter().map(|(name, scalers)| async {
+            let (cmds, res) = get_commands_and_result(
+                scalers.iter().map(|scaler| scaler.handle_event(event)),
+                "Errors occurred while handling event with all scalers",
+            )
+            .await;
+
+            let status = if cmds.is_empty() {
+                scaler_status(scalers).await
+            } else {
+                StatusInfo::compensating(
+                    "Event modified scaler {name} state, running compensating commands.",
+                )
+            };
+            if let Err(e) = self.status_publisher.publish_status(name, status).await {
+                warn!(error = ?e, "Failed to set status for scaler");
+            };
+
+            (cmds, res)
+        });
+
+        // Resolve futures, computing commands for scalers, publishing statuses, and combining any errors
+        let (commands, res) = futures::future::join_all(futs).await.into_iter().fold(
+            (vec![], Ok(())),
+            |(mut cmds, res), (mut new_cmds, new_res)| {
+                cmds.append(&mut new_cmds);
+                let res = match (res, new_res) {
+                    (Ok(_), Ok(_)) => Ok(()),
+                    (Ok(_), Err(e)) | (Err(e), Ok(_)) => Err(e),
+                    (Err(e), Err(e2)) => Err(e.context(e2)),
```

Comment:
Not sure `context` is right here. If there are two unrelated errors won't that result in `err1: err2`? Do you want to join instead?

Reply:
Is join a separate anyhow Error function? You're right they are two unrelated errors, though I haven't seen something error here yet so it's hard to say how unrelated they would be.

Reply:
No, I don't think a helper exists for this :/ I was thinking something like `=> anyhow!("{e}, {e2}")`